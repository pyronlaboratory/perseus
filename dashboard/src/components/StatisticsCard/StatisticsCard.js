import React from 'react';

// material imports
import { makeStyles } from '@material-ui/core/styles';

import CardContent from '@material-ui/core/CardContent';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles({
  root: {
    minWidth: 250,
    height: 130,
    background: 'white'
  },
  title: {
    fontSize: 12,
  },
  text: {
    fontSize: 21,
    fontWeight: 600
  }
});

const StatisticsCard = (props) => {
  const classes = useStyles();

  return (

    <Paper className={classes.root} data-testid="statistics-card" >
      <CardContent>
        <Typography className={classes.title} color="textSecondary" gutterBottom>
          {props.title}
        </Typography>
        <Typography className={classes.text} variant="h5" component="h1">
          {props.value}
        </Typography>
      </CardContent>

    </Paper>
  );
}

export default StatisticsCard
