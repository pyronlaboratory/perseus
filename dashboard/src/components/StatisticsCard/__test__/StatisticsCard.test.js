import React from 'react';
import ReactDOM from 'react-dom';

// react testing
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import StatisticsCard from './../StatisticsCard';

afterEach(cleanup);

describe('statistics card component', () => {
    it("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<StatisticsCard/>, div);
    })

    it("renders statistics card correctly", () => {
        const mockData = {
            title: "Fastest user to complete a course",
            value: "Marsh, Randy"
        }
        const { getByTestId } = render(<StatisticsCard title={mockData.title} value={mockData.value} />)
        expect(getByTestId('statistics-card')).toHaveTextContent("Marsh, Randy");
    })

    it("matches snapshot", () => {
        const mockData = {
            title: "Fastest user to complete a course",
            value: "Marsh, Randy"
        }

        const tree = renderer.create(<StatisticsCard title={mockData.title} value={mockData.value} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})