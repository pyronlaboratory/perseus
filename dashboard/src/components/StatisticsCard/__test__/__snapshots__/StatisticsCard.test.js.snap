// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`statistics card component matches snapshot 1`] = `
<div
  className="MuiPaper-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="statistics-card"
>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root makeStyles-title-2 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
    >
      Fastest user to complete a course
    </p>
    <h1
      className="MuiTypography-root makeStyles-text-3 MuiTypography-h5"
    >
      Marsh, Randy
    </h1>
  </div>
</div>
`;
